docs/byoh-instance-pre-requisites.md

```diff
@@ -0,0 +1,11 @@
+# BYOH Instance Pre-requisites
+
+The following pre-requisites must be fulfilled in order to add a Windows BYOH node.
+* The Docker container runtime must be installed on the instance.
+* The instance must be reachable by the OpenShift cluster.
+* Port 22 must be open and running [an SSH server](https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse).
+* An administrator user is present with the [provided private key](/README.md#create-a-private-key-secret) set as an authorized SSH key.
+* The hostname of the instance must:
+  * Contain only lowercase alphanumeric characters or '-'.
```

Comment:
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names
lists having '.' and upto 253 characters is allowed, should we just point to the official docs as well?

Reply:
Windows hostnames are limited to 15 characters
Since theres so few requirements I figured it would be best to just list them within this doc, that way we dont risk outside URLs changing

Reply:
Tend to agree with that, having a few bullet points here is easy to read. However, I feel should be explicit and clear that it follows the [Kubernetes' names specification](https://kubernetes.io/docs/concepts/overview/working-with-objects/names)

Reply:
Added that the specifications are from the DNS label standards of RFC 1123

Reply:
Do we need to call out the length of the hostname?